examples/vision/edsr.py

```diff
@@ -0,0 +1,320 @@
+    # If rn is less than 0.5 it returns original lr_img and hr_img
+    # If rn is greater than 0.5 it returns flipped image
+    return tf.cond(
+        rn < 0.5,
+        lambda: (lr_img, hr_img),
+        lambda: (tf.image.flip_left_right(lr_img), tf.image.flip_left_right(hr_img)),)
+
+
+def random_rotate(lr_img, hr_img):
+    """
+    Rotates Images by 90 degree
+    """
+    # Outputs random values from uniform distribution in between 0 to 4
+    rn = tf.random.uniform(shape=(), maxval=4, dtype=tf.int32)
+    # Here rn signifies number of times the image(s) are rotated by 90 degrees
+    return tf.image.rot90(lr_img, rn), tf.image.rot90(hr_img, rn)
+
+
+def random_crop(lr_img, hr_img, hr_crop_size=96, scale=4):
+    """
+    Cropping Images
+    low resolution images : 24x24
+    hight resolution images : 96x96
+    """
+    lr_crop_size = hr_crop_size // scale  # 96//4=24
+    lr_img_shape = tf.shape(lr_img)[:2]  # (height,width)
+
+    lr_width = tf.random.uniform(
+        shape=(), maxval=lr_img_shape[1] - lr_crop_size + 1, dtype=tf.int32)
+    lr_height = tf.random.uniform(
+        shape=(), maxval=lr_img_shape[0] - lr_crop_size + 1, dtype=tf.int32)
+
+    hr_width = lr_width * scale
+    hr_height = lr_height * scale
+
+    lr_img_cropped = lr_img[
+        lr_height : lr_height + lr_crop_size, lr_width : lr_width + lr_crop_size]  # 24x24
+    hr_img_cropped = hr_img[
+        hr_height : hr_height + hr_crop_size, hr_width : hr_width + hr_crop_size]  # 96x96
+
+    return lr_img_cropped, hr_img_cropped
+
+
+"""
+## Preparing tf.Data.Dataset object
+
+As the paper suggest to use RGB input patches of size 48×48 from lowres image with the
+corresponding highres patches. We augment the training data with random horizontal flips
+and 90 rotations.
+
+But we will be using 24x24 RGB input patches for lowres and corresponding highres
+patches(96x96).
+"""
+
+ds = train_cache
+
+# Applying Cropping, rotation and flipping
+ds = ds.map(lambda lr, hr: random_crop(lr, hr, scale=4), num_parallel_calls=AUTOTUNE)
+ds = ds.map(random_rotate, num_parallel_calls=AUTOTUNE)
+ds = ds.map(flip_left_right, num_parallel_calls=AUTOTUNE)
+# Batching Data
+ds = ds.batch(16)
+# Repeating Data, so that cardinality if dataset becomes infinte
+ds = ds.repeat(None)
+# prefetching allows later images to be prepared while the current image is being processed
+ds = ds.prefetch(buffer_size=AUTOTUNE)
+
+"""
+## Let's visualize a few sample images:
+"""
+
+lowres, highres = next(iter(ds))
+
+# Hight Resolution Images
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(highres[i].numpy().astype("uint8"))
+    plt.title(highres[i].shape)
+    plt.axis("off")
+
+# Low Resolution Images
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.imshow(lowres[i].numpy().astype("uint8"))
+    plt.title(lowres[i].shape)
+    plt.axis("off")
+
+
+def PSNR(sr, hr):
```

Comment:
Use a snake case name

Also use fully spelled-out variable names, not `sr` and `hr`